apps/loan/src/components/PageCrvUsdStaking/DepositWithdraw/index.tsx

```diff
@@ -0,0 +1,142 @@
+import { SubNavItem } from '@/components/PageCrvUsdStaking/components/SubNav/types'
+import { DepositWithdrawModule } from '@/components/PageCrvUsdStaking/types'
+import { useEffect } from 'react'
+import styled from 'styled-components'
+
+import useStore from '@/store/useStore'
+import { SUB_NAV_ITEMS } from '@/components/PageCrvUsdStaking/DepositWithdraw/constants'
+
+import SubNav from '@/components/PageCrvUsdStaking/components/SubNav'
+import TransactionDetails from '@/components/PageCrvUsdStaking/TransactionDetails'
+import DepositModule from '@/components/PageCrvUsdStaking/DepositWithdraw/DepositModule'
+import WithdrawModule from '@/components/PageCrvUsdStaking/DepositWithdraw/WithdrawModule'
+import DeployButton from '@/components/PageCrvUsdStaking/DepositWithdraw/DeployButton'
+import TransactionTracking from '@/components/PageCrvUsdStaking/TransactionTracking'
+
+type DepositWithdrawProps = {
+  className?: string
+}
+
+const DepositWithdraw = ({ className }: DepositWithdrawProps) => {
+  const {
+    stakingModule,
+    setStakingModule,
+    previewAction,
+    inputAmount,
+    setPreviewReset,
+    approveDepositTransaction,
+    depositTransaction,
+    depositApproval,
+    getInputAmountApproved,
+    withdrawTransaction,
+  } = useStore((state) => state.scrvusd)
+  const {
+    depositApprove: estimateGasDepositApprove,
+    deposit: estimateGasDeposit,
+    withdraw: estimateGasWithdraw,
+  } = useStore((state) => state.scrvusd.estimateGas)
+  const { lendApi, curve, curve: chainId } = useStore((state) => state)
+
+  const setNavChange = (key: SubNavItem['key']) => {
+    setStakingModule(key as DepositWithdrawModule)
+  }
+
+  const transactionInProgress =
+    (stakingModule === 'deposit' &&
```

Comment:
This comparison of the staking module smells a bit because it's repeated. I feel like these comparisons could be parts of hooks/components specific for each module.

i.e. it would be great to keep the logic out of the UI